test/integration/targets/ansible-inventory/tasks/main.yml

```diff
@@ -81,3 +81,36 @@
     that:
         - result is failed
         - '"ERROR! Could not match supplied host pattern, ignoring: invalid" in result.stderr'
+
+- name: Test toml
+  block:
+  - name: Install toml package
+    pip:
+      name:
+        - toml
+      state: present
+
+  - name: "test option: --toml with valid group name"
+    command: ansible-inventory --list --toml -i {{ role_path }}/files/valid_sample.yml
+    register: result
+
+  - assert:
+      that:
+          - result is succeeded
+
+  - name: "test option: --toml with invalid group name"
+    command: ansible-inventory --list --toml -i {{ role_path }}/files/invalid_sample.yml
+    ignore_errors: true
+    register: result
+
+  - assert:
+      that:
+          - result is failed
+          - '"ERROR! The source inventory contains a non-string key" in result.stderr'
+  # always:
+  #   - name: Uninstall toml package
+  #     pip:
+  #       name:
+  #       - toml
+  #       state: absent
+  #     ignore_errors: true
```

Comment:
Should this be uncommented? Or maybe moved to a handler since we always run handlers for tests, ensuring cleanup tasks always run even if the play fails.